ray-curator/tests/stages/deduplication/exact/test_exact_duplicates.py

```diff
@@ -0,0 +1,149 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+from typing import Literal
+
+import cudf
+import numpy as np
+import pandas as pd
+import pytest
+import ray
+
+from ray_curator.backends.experimental.ray_actor_pool import RayActorPoolExecutor
+from ray_curator.pipeline import Pipeline
+from ray_curator.stages.deduplication.exact.exact_duplicate_identification import ExactDuplicateIdentification
+from ray_curator.stages.deduplication.id_generator import CURATOR_DEDUP_ID_STR, get_id_generator_actor
+from ray_curator.tasks import FileGroupTask
+
+
+def get_original_df_with_curator_ids(
+    tasks: list[FileGroupTask], filetype: Literal["parquet", "jsonl"]
+) -> cudf.DataFrame:
+    """Get mapping from curator IDs to original IDs using IDGeneratorActor.
+    Args:
+        files: List of parquet files that were processed
+        id_column: Name of the original ID column in the data
+    Returns:
+        Dictionary mapping curator_dedup_id -> original_id
+    """
+    id_actor = get_id_generator_actor()
+    dfs = []
+    for task in tasks:
+        min_id, max_id = ray.get(id_actor.get_batch_range.remote(task.data, None))
+        df = cudf.read_parquet(task.data) if filetype == "parquet" else cudf.read_json(task.data, lines=True)
+        df[CURATOR_DEDUP_ID_STR] = np.arange(min_id, max_id + 1)
+        dfs.append(df)
+
+    return cudf.concat(dfs)
+
+
+@pytest.fixture
+def exact_dedup_data_parquet(tmp_path: Path) -> list[FileGroupTask]:
+    df1 = pd.DataFrame({"id": [1, 2, 300], "text": ["abc", "aba", "abb"]})
+    df2 = pd.DataFrame({"id": [4, -1], "text": ["aba", "abc"]})
```

Comment:
super nit.. hard to eye ball and see what the output should be since aba abb seem similar
```suggestion
    df1 = pd.DataFrame({"id": [1, 2, 300], "text": ["Small String", "Large String", "Medium String"]})
    df2 = pd.DataFrame({"id": [4, -1], "text": ["Large String", "Small String"]})
```